arangod/Transaction/Methods.cpp

```diff
@@ -1697,13 +1704,16 @@ OperationResult transaction::Methods::insertLocal(
       }
 
       replicationType = ReplicationType::LEADER;
+      if (isMMFiles && needsLock) {
+        keyLockInfo.shouldLock = true;
+      }
```

Comment:
You could reduce that to `if (needsLock) {`, because `needsLock` already implies `isMMFiles`.

Reply:
mmmh, maybe `needsLock` indirectly implies `isMMFiles`.
But in the code that is setting both variables, the assignments are completely unrelated:
```
  bool const needsLock = !isLocked(collection, AccessMode::Type::WRITE); 
  bool const isMMFiles = EngineSelectorFeature::isMMFiles();
```
So I am reluctant to change this.

Reply:
There's an assert for this a few lines above (`TRI_ASSERT(!needsLock || isMMFiles);`). If we did have `needsLock` without `isMMFiles`, several assumptions here would be broken. At the least the locks you implemented in this PR wouldn't work, so we'd get races. But I am fine with it staying this way, just wanted to note it.